### ISSUE TITLE
group should be an allowable requiredContext for menuitemcheckbox as of ARIA 1.2

### ISSUE THREAD

**Issue by dbjorge** [dequelabs/axe-core#2736]
Expectation: As of [ARIA 1.2](https://www.w3.org/TR/wai-aria-1.2/#menuitemcheckbox) (specifically, PR https://github.com/w3c/aria/issues/737), `group` has become an acceptable required context role for the `menuitemcheckbox` role; axe-core should recognize this now that [it's picking up other ARIA 1.2 role relationship updates](https://github.com/dequelabs/axe-core/issues/2107).

Actual: `lib/standards/aria-roles.js` hasn't been updated to account for that yet, so axe 4.1.1 still reports an `aria-required-parent` failure for the button in this snippet ([codepen](https://codepen.io/dbjorge/full/WNGyWmd)):

```html
<div role="group">
  <button role="menuitemcheckbox">test element</button>
</div>
```

Motivation: false positive in aria-required-parent rule

<pre><code>
axe-core version: 4.1.1
axe-webdriver, extension or other integration version: accessibility insights for web 0.23.0

Browser and Assistive Technology versions: Microsoft Edge stable (Version 87.0.664.66 (Official build) (64-bit)), no AT in use
</code></pre>

**Comment by straker** [comment]
We've been having some discussion of this in the PR that was opened for it. There's some issues this causes for screen reader so we're determining what should happen.
